Stratis.Bitcoin.Features.RPC/RpcSettings.cs

```diff
@@ -58,58 +58,69 @@ public void Load(NodeSettings nodeSettings)
             var config = nodeSettings.ConfigReader;
 
             this.Server = config.GetOrDefault<bool>("server", false);
-            if (!this.Server)
-                return;
+            this.RPCPort = nodeSettings.Network.RPCPort;
+
+            if (this.Server)
+            {
+                this.RpcUser = config.GetOrDefault<string>("rpcuser", null);
+                this.RpcPassword = config.GetOrDefault<string>("rpcpassword", null);
+
+                var defaultPort = config.GetOrDefault<int>("rpcport", nodeSettings.Network.RPCPort);
+                this.RPCPort = defaultPort;
+
+                try
+                {
+                    this.AllowIp = config
+                        .GetAll("rpcallowip")
+                        .Select(p => IPAddress.Parse(p))
+                        .ToList();
+                }
+                catch (FormatException)
+                {
+                    throw new ConfigurationException("Invalid rpcallowip value");
+                }
+            }
+
+            this.callback?.Invoke(this);
```

Comment:
I find the fact that we have this callback invoked in the middle of the method quite confusing.
Can we do something like this:
1. get values from config
2. invoke callback
3. check validity of settings

Reply:
I agree. I've re-factored the code by splitting it into additional private methods.